### PR TITLE
Add unary call to generation tests

### PR DESCRIPTION
**Pull request by buildbreaker**
Noticed that this behavior hasn't been working exactly as expected for certain edge cases due to the exclusion from the `subList()` call

### CODE REVIEW

File: protoc-gen-connect-kotlin/src/main/kotlin/build/buf/protocgen/connect/Generator.kt [hunk 1]
```diff
@@ -307,20 +310,20 @@ class Generator : CodeGenerator {
...
+        // becomes ["EmptyMessage", "InnerMessage"].
         val names = getClassName(descriptor)
             .removePrefix(packageName)
             .removePrefix(".")
             .split(".")
         // Case when there is a nested entity.
         // e.g Nested message definitions and messages within "*OuterClass.java".
         if (names.size > 1) {
-            return ClassName(packageName, names.first(), *names.subList(1, names.size - 1).toTypedArray())
+            return ClassName(packageName, names.first(), *names.subList(1, names.size).toTypedArray())
```
[the remark this answers is not in the document]
Comment (buildbreaker) [reply protoc-gen-connect-kotlin/src/main/kotlin/build/buf/protocgen/connect/Generator.kt:326]: Yeah, I gave it a better thought and we need to actually pass the objects through to the generated methods to exercise this problematic case...

[the remark this answers is not in the document]
Comment (buildbreaker) [reply protoc-gen-connect-kotlin/src/main/kotlin/build/buf/protocgen/connect/Generator.kt:326]: The tests are now more beefed up
